scripts/puritan.lic

```diff
@@ -313,28 +313,41 @@ module Puritan
     end
   end
 
-  def self.outcome_of_1004(gem)
-    waitcastrt?
-    waitrt?
-    Spell.lock_cast()
-    prep_1004()
-    fput("sing ##{gem.id}")
-    if wait_for_own_song().eql?(:ok)
-      result = matchwait(
-        #"gem becomes more perfect",
-        #"improves somewhat",
+  def self.select_outcome_from_1004_output()
+    while line=get
+      if line =~ Puritan::Outcomes::ALL
+        outcome = line
+      end
+      if Puritan.stop_processing_1004_result_on?(line)
+        outcome ||= line
+        return outcome
+      end
+    end
+  end
+
+  def self.stop_processing_1004_result_on?(line)
+    [
         "turn as the very essence",
         "sparkles a little more",
         "shatter",
-        "crack",
         "must be holding",
         "what were you",
-        "appearing smoother and more pure",
-        "cannot be", 
+        "cannot be",
         "Sing Roundtime",
         "Wait",
         'Spell Hindrance',
-        "song misfires")
+        "song misfires"
+    ].any? { |s| line.include?(s) }
+  end
+
+  def self.outcome_of_1004(gem)
+    waitcastrt?
+    waitrt?
+    Spell.lock_cast()
+    prep_1004()
+    fput("sing ##{gem.id}")
+    if wait_for_own_song().eql?(:ok)
+      result = Puritan.select_outcome_from_1004_output()
       Spell.unlock_cast()
```

Comment:
Unrelated to this PR and I haven't noticed any issues, but it seems like we should do this whether or not `wait_for_own_song()` returns `:ok`